content/en/blog/_posts/2021-04-16-volume-health-monitoring-alpha.md

```diff
@@ -0,0 +1,97 @@
+
+## How do I use Volume Health on my Kubernetes Cluster?
+
+To use the Volume Health feature, first make sure the CSI driver you are using supports this feature. Refer to this [CSI drivers doc](https://kubernetes-csi.github.io/docs/drivers.html) to find out which CSI drivers support this feature.
+
+To enable Volume Health Monitoring from the node side, this alpha feature gate `CSIVolumeHealth` needs to be enabled.
+
+If a CSI driver supports the Volume Health Monitoring feature from the controller side, events regarding abnormal volume conditions will be recorded on PVCs.
+
+If a CSI driver supports the Volume Health Monitoring feature from the controller side, user can also get events regarding node failures if the `enable-node-watcher` flag is set to true when deploying the External Health Monitor Controller. When a node failure event is detected, an event will be reported on the PVC to indicate that pods using this PVC are on a failed node.
+
+If a CSI driver supports the Volume Health Monitoring feature from the node side, events regarding abnormal volume conditions will be recorded on pods using the PVCs.
+
+## As a storage vendor, how do I add support for volume health to my CSI driver?
+
+Volume Health Monitoring includes two parts:
+* An External Volume Health Monitoring Controller monitors volume health from the controller side.
+* Kubelet monitors volume health from the node side.
+
+For details, see the [CSI spec](https://github.com/container-storage-interface/spec/blob/master/spec.md) and the [Kubernetes-CSI Driver Developer Guide](https://kubernetes-csi.github.io/docs/volume-health-monitor.html).
+
+There is a sample implementation for volume health in [CSI host path driver](https://github.com/kubernetes-csi/csi-driver-host-path).
+
+### Controller Side Volume Health Monitoring
+
+To learn how to deploy External Volume Health Monitoring Controller, see details in this [doc](
+https://kubernetes-csi.github.io/docs/external-health-monitor-controller.html).
+
+The External Health Monitor Controller calls either `ListVolumes` or `ControllerGetVolume` CSI RPC and reports VolumeConditionAbnormal events with messages on PVCs if abnormal volume conditions are detected. Only CSI drivers with `LIST_VOLUMES` and `VOLUME_CONDITION` controller capability or `GET_VOLUME` and `VOLUME_CONDITION` controller capability support Volume Health Monitoring in the external controller.
+
+To implement the volume health feature from the controller side, a CSI driver MUST add support for the new controller capabilities.
+
+If a CSI driver supports `LIST_VOLUMES` and `VOLUME_CONDITION` controller capabilities, it MUST implement controller RPC `ListVolumes` and report the volume condition in the response.
+
+If a CSI driver supports `GET_VOLUME` and `VOLUME_CONDITION` controller capability, it MUST implement controller PRC `ControllerGetVolume` and report the volume condition in the response.
```

Comment:
```suggestion
To implement the volume health feature from the controller side, a CSI driver **must** add support for the new controller capabilities.

If a CSI driver supports `LIST_VOLUMES` and `VOLUME_CONDITION` controller capabilities, it **must** implement controller RPC `ListVolumes` and report the volume condition in the response.

If a CSI driver supports `GET_VOLUME` and `VOLUME_CONDITION` controller capability, it **must** implement controller PRC `ControllerGetVolume` and report the volume condition in the response.
```
We prefer bold to capitals for emphasis. Not essential.

Reply:
Done